mailbox/plugin/spamassassin/src/main/java/org/apache/james/mailbox/spamassassin/SpamAssassinListener.java

```diff
@@ -107,33 +107,32 @@ private void handleAdded(Event event, MailboxSession session, Added addedEvent)
             Mailbox mailbox = mapperFactory.getMailboxMapper(session).findMailboxById(addedEvent.getMailboxId()).block();
             MessageMapper messageMapper = mapperFactory.getMessageMapper(session);
 
-            List<InputStream> contents = MessageRange.toRanges(addedEvent.getUids())
-                .stream()
+            Flux<InputStream> contents = Mono.fromCallable(() -> MessageRange.toRanges(addedEvent.getUids()))
+                .flatMapIterable(Function.identity())
                 .flatMap(range -> retrieveMessages(messageMapper, mailbox, range))
```

Comment:
This listener mixes retrieval by MessageId, and by MailboxId + Uid.

We could plan to retrieve everything by MessageId IMO

Reply:
I do not get it, can you explicit it please

Reply:
messageIdMapper.find(addedEvent.getMessageIds())

Reply:
Impossible due to MaildirMailboxMapper